### PR TITLE
Add between parser combinator to Data.String.Parser

### PR DESCRIPTION
**Pull request by justjoheinz**
# Description

between - an additional parser combinator in Data.String.Parser (contrib).

### CODE REVIEW

File: libs/contrib/Data/String/Parser.idr [hunk 1]
```diff
@@ -406,3 +406,14 @@ export
...
+
+||| Run the parser p between the opening parser `o` and the closing parser `c`,
+||| returning the result of p.
+export
+between : Monad m => ParseT m o -> ParseT m c -> ParseT m p -> ParseT m p
+between o c p = do
+  _ <- o
+  result <- p
+  _ <- c
+  pure result
```
Comment (gallais) [inline libs/contrib/Data/String/Parser.idr:419]:
```suggestion
between o c p = id <$ o <*> p <* c
```

Comment (justjoheinz) [reply libs/contrib/Data/String/Parser.idr:419]: I have a bit of a hard time understanding the suggestion - is it equivalent to 
```
between o c p = o *> p *< c
```
?

Comment (gallais) [reply libs/contrib/Data/String/Parser.idr]: The core idea is to never mix left & right pointed combinators. Only have `f arg1 arg2 arg3` with `<*` before unused arguments and `<*>` before used ones.

**Comment by andrevidela** [comment]
with AOC upon us this is more useful than ever

**Pull request merged**